### PR TITLE
feat(e2e2): run tests also against secondary NS

### PR DESCRIPTION
**Pull request by nils-wisiol**
This is part of a larger effort to run e2e2 tests against desec-ns.

### CODE REVIEW

File: test/e2e2/spec/test_dyndns.py
Comment (peterthomassen) [inline test/e2e2/spec/test_dyndns.py:56]: no `return` needed

**Review by peterthomassen — COMMENTED** [review]
Did a semi-in-depth review, looks reasonable so far. -- Do you want to discuss something specific? Do you want an approval-grade review now, or will more changes come?

File: test/e2e2/spec/test_dyndns.py
Comment (nils-wisiol) [reply test/e2e2/spec/test_dyndns.py:56]: That's what I initially thought, too. But: This function is called in conftest.py:484 by `assert assertion(**(assertion_kwargs or dict()))`. No return statement leads to `assert None`, which of course fails.

In most cases, the `assertion` function given to `assert_all_ns` returns a Boolean value (actually, something that is truthy or falsey) and so not assert themselves anything. But here, I used the two legacy assert statements because (a) there were already there and (b) it seemed cleaner than `and`-ing them.

Comment (peterthomassen) [reply test/e2e2/spec/test_dyndns.py:56]: I think `assert_eventually` should then have a flag to decide whether to wrap its first argument in `_assert` or not.

**Comment by nils-wisiol** [comment]
More changes here will only come if made necessary by desec-io/desec-ns#58.

**Review by peterthomassen — APPROVED** [review]
lgtm, pending [the desec-ns PR.](https://github.com/desec-io/desec-ns/pull/58).

File: test/e2e2/spec/test_dyndns.py
Comment (nils-wisiol) [reply test/e2e2/spec/test_dyndns.py:51]: `rrs_dns` will only ever be looked at at `rrs_dns[qtype]`, so filling the other value is unnecessary.

File: test/e2e2/spec/test_dyndns.py
Comment (nils-wisiol) [reply test/e2e2/spec/test_dyndns.py:50]: No because the call is also inside the for loop:

```shell
>>> for i in range(2):
...   def callme():
...     print(i)
...   callme()
... 
0
1
```

File: test/e2e2/conftest.py
Comment (nils-wisiol) [reply test/e2e2/conftest.py:342]: Yes, changes to nslord are made before the api request returns and can thus be expected to be visible immediately.

File: test/e2e2/spec/test_dyndns.py
Comment (nils-wisiol) [reply test/e2e2/spec/test_dyndns.py]: I've addressed this in a fixup commit

File: test/e2e2/spec/test_dyndns.py
Comment (nils-wisiol) [reply test/e2e2/spec/test_dyndns.py]: I've addressed this in a fixup commit

File: test/e2e2/spec/test_dyndns.py
Comment (peterthomassen) [reply test/e2e2/spec/test_dyndns.py]: Oh yes. I had written that assuming that you would fix then typo `qtype` to `qt`, which would then make the call independent of the iteration variables. But you refactored, so no longer applies.

**Review by peterthomassen — APPROVED** [review]
lgtm

**Pull request merged**
